linera-chain/src/chain.rs

```diff
@@ -494,14 +495,28 @@ where
         let balance = self.execution_state.system.balance.get_mut();
 
         balance.try_add_assign(credit)?;
-        Self::sub_assign_fees(balance, pricing.certificate_price())?;
-        Self::sub_assign_fees(balance, pricing.storage_price(&block.incoming_messages)?)?;
-        Self::sub_assign_fees(balance, pricing.storage_price(&block.operations)?)?;
+        sub_assign_fees(balance, policy.certificate_price())?;
+        sub_assign_fees(
+            balance,
+            policy.storage_bytes_written_price_raw(&block.incoming_messages)?,
+        )?;
+        sub_assign_fees(
+            balance,
+            policy.storage_bytes_written_price_raw(&block.operations)?,
+        )?;
 
         let mut messages = Vec::new();
         let mut message_counts = Vec::new();
-        let available_fuel = pricing.remaining_fuel(*balance);
-        let mut remaining_fuel = available_fuel;
+        let maximum_bytes_read = policy.maximum_bytes_read;
+        let maximum_bytes_written = policy.maximum_bytes_written;
+        let mut runtime_track = RuntimeTracker {
+            used_fuel: 0,
+            num_reads: 0,
+            bytes_read: 0,
+            bytes_written: 0,
+            maximum_bytes_read,
```

Comment:
Why copy elements of the policy inside the tracker?

Reply:
I see. Perhaps different names could help. (We're conflating the policy `maximum_bytes_read_per_block` with the tracker `maximum_bytes_left_to_read`.)

Reply:
Thanks, renamed.